wgengine/userspace.go

```diff
@@ -109,6 +109,7 @@ type userspaceEngine struct {
 	trimmedDisco        map[tailcfg.DiscoKey]bool // set of disco keys of peers currently excluded from wireguard config
 	sentActivityAt      map[netaddr.IP]*int64     // value is atomic int64 of unixtime
 	destIPActivityFuncs map[netaddr.IP]func()
+	networkMapCallbacks map[*networkMapCallbackHandle]NetworkMapCallback
```

Comment:
probably want to move this down into the "mu sync.Mutex" section a few lines down

it's unusual (and too complicated) to have types with multiple mutexes in them. This is some nasty code in that regard, sorry.

but the be sure to acquire mu whenever you access this map.